src/adagrams.js

```diff
@@ -1,8 +1,162 @@
+
+    const lettersInHand = [];
+    // Method from Stack Overflow here: https://stackoverflow.com/questions/5915096/get-random-item-from-javascript-array
+    // Math.random returns a floating point between 0-1, multiplies that by the length of the array, and Math.floor rounds that down to determine a 'random' index - is it really random?  According to some people, no.  But we'll take it for our first go at a randomizer. 
+    for(let i = 0; i < 10; i++) {
+      let randomIndex = Math.floor(Math.random()*letters.length);
+      let letter = letters[randomIndex];
+      
+      letters.splice(randomIndex, 1);
+
+      lettersInHand.push(letter.toUpperCase());
+    };
+
+    return lettersInHand;
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    const lettersInHandCopy = lettersInHand.map((letter) => letter);
+  
+    const splitInput = input.split('');
+        
+    for (let i = 0; i < splitInput.length; i++) {
+      if (lettersInHandCopy.includes(splitInput[i])) {
+        const index = lettersInHandCopy.indexOf(splitInput[i]);
+        lettersInHandCopy.splice(index, 1);
+      } else {
+        return false;
+      };
+  
+    };
+    return true;
+  },
+
+  scoreWord(word) {
+    let score = 0;
+  
+    if (word.length > 6) {
+      score += 8;
+    };
+  
+    const wordArray = word.toUpperCase().split('');
+  
+    for (const char of wordArray) {
+      switch(char) {
+        case 'A':
+        case 'E': 
+        case 'I': 
+        case 'O': 
+        case 'U': 
+        case 'L': 
+        case 'N': 
+        case 'R': 
+        case 'S':
+        case 'T':
+          score += 1;
+          break;
+        case 'D':
+        case 'G':
+          score += 2;
+          break;
+        case 'B':
+        case 'C': 
+        case 'M':
+        case 'P':
+          score += 3;
+          break;
+        case 'F': 
+        case 'H': 
+        case 'V': 
+        case 'W':
+        case 'Y':
+          score += 4;
+          break;
+        case 'K':
+          score += 5;
+          break;
+        case 'J': 
+        case 'X':
+          score += 8;
+          break;
+        case 'Q':
+        case 'Z':
+          score += 10;
+          break;
+      }
+    };
+    return score;
+  },
+
+  wordScores(words) {
+    const wordsAndScores = {};
+    
+     words.forEach(word => {
+      wordsAndScores[word] = this.scoreWord(word);
+    });
+
+    return wordsAndScores;
+  },
+
+  tieBreaker(words) {
+    const tenLetterWords = [];
+    const tieScoreWords = [];
+    let winner = null;
+
+    const scores = this.wordScores(words);
+    const entries = Object.entries(scores);
+
+    for (let [word, score] of entries) {
```

Comment:
since you're looking at each word in `entries`, it may be worth using a forEach loop in this case